Userland/Libraries/LibIDL/Types.cpp

```diff
@@ -214,9 +214,9 @@ int EffectiveOverloadSet::distinguishing_argument_index()
 
 void EffectiveOverloadSet::remove_all_other_entries()
 {
-    m_items.remove_all_matching([this](auto const& item) {
-        return &item != m_last_matching_item;
-    });
```

Comment:
you could also have fixed `Vector::remove_all_matching()` by making it recurse through the list backwards, but I agree your new implementation for `EffectiveOverloadSet::remove_all_other_entries()` is still better.